File: week2/day2/week2-day2/week2-day2/Farm.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace week2_day2
{
    class Farm
    {
        public List<Animal> animalList=new List<Animal>();
        public int slot;

        public Farm()
        {
            this.slot = 100;
        }

        public void breed(Animal a)
        {
            if (this.slot > 0)
            {
                this.animalList.Add(a);
                slot--;
            }

        }

        public void Slaughter()
        {
            int least =int.MaxValue;
            int remove=0;
            for(int i=0; i<this.animalList.Count;i++)
            {
                if (this.animalList[i].hunger < least)
                {
                    least = this.animalList[i].hunger;
                    remove = i;
                }
            }
            animalList.RemoveAt(remove);
        }
    }
}

File: DSA-2019/week-02/day-1/farm.py
from animal import Animal


class Farm():
    def __init__(self, places, animals=[]):
        self.animals = animals
        self.places = places

    def add_animal(self, animal: Animal):
        self.animals.append(animal)

    def breed(self, animal: Animal):
        if(self.places>0):
            self.animals.append(animal)
        else:
            raise Exception("no slot avaliable")

    def slaughter(self):
        newlist = sorted(self.animals, key=lambda x: x.hunger)
